web/models/github_app_installation.ex

```diff
@@ -21,47 +24,31 @@ defmodule CodeCorps.GithubAppInstallation do
   end
 
   @doc """
-  Builds a changeset based on the `struct` and `params`.
+  Changeset used to create a GithubAppInstallation record from CodeCorps
   """
   def create_changeset(struct, params \\ %{}) do
     struct
-    |> changeset(params)
+    |> cast(params, [:project_id, :user_id])
+    |> put_change(:state, "unprocessed")
+    |> put_change(:origin, "codecorps")
+    |> validate_required([:project_id, :user_id])
+    |> assoc_constraint(:project)
+    |> assoc_constraint(:user)
```

Comment:
If I'm following things correctly, this changeset only gets called when creating from the client, in which case, the `state` and the `origin` need to have specific values. No need for a state_machine anymore, or anything of the sort. The `github_id` field is not set from here either.